### ISSUE TITLE
fastrtps (Fast-DDS) version 2.3.x does not have ContentFilteredTopic required by humble branch

### ISSUE THREAD

**Issue by henriksod** [ros2/rmw_fastrtps#700]
## Bug report

**Required Info:**

- Version or commit hash:
  - c6efdc9fd47cd50043535a2bff27a5c402728bab
 
#### Steps to reproduce issue

Incorrect fastrtps version is listed in the humble branch of this repo. It is listed as `2.3`:
https://github.com/ros2/rmw_fastrtps/blob/c6efdc9fd47cd50043535a2bff27a5c402728bab/rmw_fastrtps_shared_cpp/CMakeLists.txt#L47

fastrtps `2.3.x` does not have `fastdds/dds/topic/ContentFilteredTopic.hpp`, as is required here:
https://github.com/ros2/rmw_fastrtps/blob/c6efdc9fd47cd50043535a2bff27a5c402728bab/rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_subscriber_info.hpp#L32

Here is a link to fastrtps `2.3.6` under `include/fastdds/dds/topic`:
https://github.com/eProsima/Fast-DDS/tree/2.3.6/include/fastdds/dds/topic

This feature was added in fastrtps `2.5.0`.
https://github.com/eProsima/Fast-DDS/releases/tag/v2.5.0

An additional issue is with the following call to `eprosima::fastdds::dds::DataReader::get_unread_count`:
https://github.com/ros2/rmw_fastrtps/blob/c6efdc9fd47cd50043535a2bff27a5c402728bab/rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_service_info.hpp#L206

The boolean function argument was added in `2.7.1`:
https://github.com/eProsima/Fast-DDS/commit/9586068ba852a31a74f26109fc425f7e79a8c5c1

#### Expected behavior

The correct minimum fastrtps version to be listed in `rmw_fastrtps_shared_cpp/CMakeLists.txt`

#### Actual behavior

An incompatible minimum version of fastrtps is currently listed under `rmw_fastrtps_shared_cpp/CMakeLists.txt` in the humble branch.

#### Implementation considerations

Replace `2.3` L47 and L48 with `2.7.1`:
https://github.com/ros2/rmw_fastrtps/blob/c6efdc9fd47cd50043535a2bff27a5c402728bab/rmw_fastrtps_shared_cpp/CMakeLists.txt

**Comment by clalancette** [comment]
Please feel free to open a pull request to update that; it would be much appreciated.

**Comment by fujitatomoya** [comment]
> Replace 2.3 L47 and L48 with 2.7.1

https://github.com/eProsima/Fast-DDS/pull/2825 has been backported to `2.6.x` branch with https://github.com/eProsima/Fast-DDS/pull/2843, which is available on `2.6.2` or later.

So i think it should be update into `2.6.2`.

**Comment by fujitatomoya** [comment]
This has been addressed with Humble patch release https://github.com/ros2/rmw_fastrtps/commit/4b0f30f15fad0e1a9824db0bf54dbbaa5085fa5d

```bash
root@tomoyafujita:~# dpkg -l | grep rmw-fastrtps
ii  ros-humble-rmw-fastrtps-cpp                       6.2.4-1jammy.20230919.211331            amd64        Implement the ROS middleware interface using eProsima FastRTPS static code generation in C++.
ii  ros-humble-rmw-fastrtps-shared-cpp                6.2.4-1jammy.20230919.205954            amd64        Code shared on static and dynamic type support of rmw_fastrtps_cpp.
```

and as we described above, it uses, `v2.6.2`

https://github.com/ros2/rmw_fastrtps/blob/4b0f30f15fad0e1a9824db0bf54dbbaa5085fa5d/rmw_fastrtps_shared_cpp/CMakeLists.txt#L47-L48

I will go ahead to close this issue.
